### PR TITLE
feat(web): quality assurance

### PR DESCRIPTION
**Pull request by m8vago**
- Add posthog for quality assurance purposes.
- Add captures for common elements.
- Add env variables for disable QA.

### CODE REVIEW

File: web/crux-ui/src/components/projects/versions/deployments/deployment-view-list.tsx [hunk 1]
```diff
@@ -28,6 +28,7 @@ const DeploymentViewList = (props: DeploymentViewListProps) => {
           selected: deployInstances,
           onAllChange: actions.onAllInstancesToggled,
           onChange: actions.onInstanceSelected,
+          qaLabel: `instance`,
```
Comment (robot9706) [inline web/crux-ui/src/components/projects/versions/deployments/deployment-view-list.tsx:31]: Backtick unnecessary.

File: web/crux-ui/src/elements/dyo-button.tsx [hunk 2]
```diff
@@ -76,13 +93,14 @@ const DyoButton = (props: DyoButtonProps) => {
...
       )}
+      onClick={onClick}
     >
       {children}
     </button>
   )
 
   return href ? (
-    <Link className="inline-block" href={href}>
+    <Link className="inline-block" href={href} onClick={() => sendQAEvent()}>
```
Comment (robot9706) [inline web/crux-ui/src/elements/dyo-button.tsx:103]: Simply `onClick={sendQAEvent}` should work too.

**Review by robot9706 — COMMENTED** [review]
Minor comments, haven't tested it yet.

**Comment by gerimate** [comment]
Can you please add this link directing to the docs pls?
- https://docs.dyrector.io/learn-more/software-quality-assurance-sqa

File: web/crux-ui/src/hooks/use-posthog.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  }
+
+const onLoaded = (ph: PostHog) => {
+  if (process.env.NODE_ENV === 'development') {
+    ph.debug()
+  }
+}
+
+if (!isServerSide()) {
+  posthog.init('phc_wc4nEUy7HYW8Elf8G9jiccvl2ZYt8pVts8NVBRMPzHu', {
```
Comment (polaroi8d) [inline web/crux-ui/src/hooks/use-posthog.ts:36]: Is this not a security leak?

**Review by polaroi8d — CHANGES_REQUESTED** [review]
In general I think it's not QA, it's Analytics.

File: web/crux-ui/src/elements/dyo-link.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    qaLabel: string
+    children: React.ReactNode
+  } & React.RefAttributes<HTMLAnchorElement>
+
+const DyoLink = (props: DyoLinkProps) => {
+  const { qaLabel, onClick: propsOnClick, ...forwardedProps } = props
+
+  const sendQAEvent = useCallback(() => {
+    sendQAClickEvent({
+      elementType: 'a',
```
[the remark this answers is not in the document]
Comment (robot9706) [reply web/crux-ui/src/elements/dyo-link.tsx:16]: `<a>` as the html anchor.

**Review by robot9706 — APPROVED** [review]

**Pull request merged**